### PR TITLE
MM-36591 Custom Statuses: Add custom status to mention autocomplete

### PR DESCRIPTION
**Pull request by manojmalik20**
#### Summary

- Fixed MM-36591 RN: Custom Statuses: Add custom status to mention autocomplete
- Changed the default behavior of autocomplete item UI in case of long display names. Long display names will be truncated to show the username.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-36591

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [ ] Includes text changes and localization file updates

#### Device Information
This PR was tested on: 

- iPhone 12 Simulator - iOS 14.4
- Android Emulator - Pixel_3a_API_30_x86 - Android 10

#### Screenshots

image

image

image

#### Release Note

```release-note
- Fixed MM-36591 RN: Custom Statuses: Add custom status to mention autocomplete
- Changed the default behavior of autocomplete item UI in case of long display names. Long display names will be truncated to show the username.
```

### CODE REVIEW

**Comment by matthewbirtch** [comment]
Thanks @manojmalik20. It's looking great. The only thing I noticed is that we are using the ellipses differently here than we are in the production app. Is there a reason for that? I would prefer the ellipses be used at the end always rather than in the middle.
<table>
<tr>
<td>
<p><strong>Production App</strong></p>

</td>
<td>
<p><strong>This PR</strong></p>

</td>
</tr>
</table>

**Comment by manojmalik20** [comment]
@matthewbirtch Actually in the production app, if there is a really long name then the username isn't visible at all. So I've changed that behavior. If a user has a really long name, it will be truncated with ellipses to show the username also.

**Comment by matthewbirtch** [comment]
> @matthewbirtch Actually in the production app, if there is a really long name then the username isn't visible at all. So I've changed that behavior. If a user has a really long name, it will be truncated with ellipses to show the username also.

Right, I understand that. However, this approach favours the username over the display name. I would expect it to favour the display name. It's also strange to have an ellipses in the middle.

I don't think this change was expected as part of this ticket, so would you be able to revert that change?

**Comment by manojmalik20** [comment]
@matthewbirtch I've reverted that change.

**Comment by matthewbirtch** [comment]
@manojmalik20 now I'm not seeing the elipses at all. The text now just gets cut off (screenshot below)

**Comment by manojmalik20** [comment]
@matthewbirtch I'm not able to reproduce this on either Android or IOS. 

Simulator Screen Shot - iPhone 12 - 2021-07-02 at 05 11 10

Can you tell me the version you are using for both ios and android?

**Comment by matthewbirtch** [comment]
> Can you tell me the version you are using for both ios and android?

@manojmalik20 I have iOS 14.6 on my iPhone. 

The screenshot you showed above truncates the display name and then cuts off the username (you can actually see it getting cut off on your screenshot). The truncation should be at the end of both elements no matter what. If the display name is longer than what is visible, then we should see the ellipses at the end of the display name and we wouldn't see the username at all. If the display name is short, but the username is longer than what's visible, the ellipses would show at the end of the username

**Comment by manojmalik20** [comment]
@matthewbirtch I've fixed the username cutting off issue.


Simulator Screen Shot - iPhone 12 - 2021-07-07 at 05 59 25

**Comment by matthewbirtch** [comment]
@manojmalik20 the build failed. The screenshots look good though

**Comment by manojmalik20** [comment]
@matthewbirtch Can you please re-trigger the build and add more reviewers?

**Comment by manojmalik20** [comment]
@matthewbirtch The build is failing at the Checkout code step. Do you know what might be the reason for that?

**Comment by matthewbirtch** [comment]
> @matthewbirtch The build is failing at the Checkout code step. Do you know what might be the reason for that?

I'm afraid I'm not well versed in the build processes. @enahum can you help out?

**Comment by enahum** [comment]
The CI was having an outage yesterday, let's try again

**Comment by manojmalik20** [comment]
@enahum The build failed again in the `Getting Javascript Dependencies` step due to socket timeout. Maybe try again?

**Comment by enahum** [comment]
Build is ready

**Review by matthewbirtch — APPROVED** [review]
I think we're good @manojmalik20

**Comment by matthewbirtch** [comment]
@esethna can you help assign appropriate engineering reviewers?

**Review by enahum — APPROVED** [review]

File: app/components/autocomplete/at_mention_item/at_mention_item.tsx [hunk 1]
```diff
@@ -121,28 +122,37 @@ const AtMentionItem = ({firstName = '', isBot, isCurrentUser, isGuest, isShared,
...
+                                />
+                            )}
+                            {` @${username}`}
+                        </Text>
                     </Text>
                 </View>
+                {isCustomStatusEnabled && !isBot && (
+                    <CustomStatusEmoji
+                        userID={userId}
+                        style={{marginLeft: 4}}
```
Comment (anurag6713) [inline app/components/autocomplete/at_mention_item/at_mention_item.tsx:153]: 1. Can we move this out from inline?
2. Add the same style to Shared channel's `ChannelIcon` component?

File: app/components/autocomplete/at_mention_item/at_mention_item.tsx [hunk 2]
```diff
@@ -50,24 +52,23 @@ const getStyleFromTheme = makeStyleSheetFromTheme((theme: Theme) => {
         },
         rowInfo: {
             flexDirection: 'row',
-            flex: 1,
```
Comment (anurag6713) [inline app/components/autocomplete/at_mention_item/at_mention_item.tsx:53]: Can you make the shared channel Icon visible in code and fix this?  
Screenshot 2021-07-09 at 11 07 55

**Review by anurag6713 — CHANGES_REQUESTED** [review]

**Comment by manojmalik20** [comment]
@anurag6713 This is ready for re-review.

**Review by anurag6713 — APPROVED** [review]

**Comment by anurag6713** [comment]
@manojmalik20 LGTM! Thank you 🙂

**Review by jgilliam17 — APPROVED** [review]
Thank you @manojmalik20 
Tested, looks good to merge.
- Verified custom status displays on mention autocomplete, and also long display names are truncated properly on the right.

**Pull request merged**